vcr/holder/wallet.go

```diff
@@ -121,21 +121,16 @@ func (h wallet) BuildPresentation(ctx context.Context, credentials []vc.Verifiab
 }
 
 func (h wallet) resolveSubjectDID(credentials []vc.VerifiableCredential) (*did.DID, error) {
-	type credentialSubject struct {
-		ID did.DID `json:"id"`
-	}
 	var subjectID did.DID
 	for _, credential := range credentials {
-		var subjects []credentialSubject
-		err := credential.UnmarshalCredentialSubject(&subjects)
-		if err != nil || len(subjects) != 1 {
-			return nil, errors.New("not all VCs contain credentialSubject.id")
+		sid, err := credential.SubjectDID()
+		if err != nil {
+			return nil, err
 		}
-		subject := subjects[0]
```

Comment:
this is changed behavior since it now allows multiple subjects as long as they as have the same Id? (same in issuer) Now less strict so probably fine.

Reply:
Yes, but it will ultimately only use 1 of them or fail in some place else. This is, after all, just storage.